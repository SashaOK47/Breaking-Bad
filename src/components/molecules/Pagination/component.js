// import PropTypes from "prop-types";
import style from "./styles.module.scss";
import arrow from "../../../assets/icons/arrow.svg";

const Pagination = ({
  arrOfCurrButtons,
  currentPage,
  handlerBtnNext,
  handlerBtnPrev,
  handleClick,
}) => {
  return (
    <div className={style.pagination}>
      <div className={style.paginationList}>
        {arrOfCurrButtons.map((item, index) => {
          let cls = style.paginationItem;
          if (currentPage == item) {
            cls += ` ${style.active}`;
          } else if (item === "...") {
            cls += ` ${style.disabled}`;
          }
          return (
            <button key={index} className={cls} onClick={handleClick}>
              {item}
            </button>
          );
        })}
      </div>
      <button className={style.btnPrev} onClick={handlerBtnPrev}>
        <img src={arrow} alt="Button Prev" />
      </button>
      <button className={style.btnNext} onClick={handlerBtnNext}>
        <img src={arrow} alt="Button Next" />
      </button>
    </div>
  );
};
export default Pagination;
