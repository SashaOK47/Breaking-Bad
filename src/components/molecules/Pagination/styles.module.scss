@import "../../../assets/styles/main.scss";
.pagination {
  display: flex;
  @include tablet {
    margin-bottom: 1.5rem;
  }
  .paginationList {
    display: flex;
    align-items: center;
    button {
      @include filterBtn;
      padding: 0.25rem 0.75rem;
      @include tablet {
        padding: 0.25rem 0.5rem;
      }
      &.disabled {
        cursor: no-drop;
        padding: 0;
        &:focus-visible {
          box-shadow: none;
        }
      }
    }
  }
  .btnPrev,
  .btnNext {
    background: none;
    &:focus-visible {
      @include boxShadowFocus;
    }
  }
  .btnPrev {
    margin-left: 2rem;
    @include tablet {
      margin-left: 1rem;
    }
  }
  .btnNext {
    margin-left: 1.25rem;
    transform: rotate(180deg);
    @include tablet {
      margin-left: 1rem;
    }
  }
}
