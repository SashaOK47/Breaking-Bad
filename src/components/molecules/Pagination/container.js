import Pagination from "./component";

const PaginationContainer = () => {
  return <Pagination />;
};
export const container = PaginationContainer;
