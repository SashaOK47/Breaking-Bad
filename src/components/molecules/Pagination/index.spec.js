import React from "react";
import Pagination from "./component";

const arrOfCurrButtons = [1, 2, 3, "..."];
const currentPage = 1;
const handlerBtnNext = jest.fn();
const handlerBtnPrev = jest.fn();
const handleClick = jest.fn();

const setUp = (props) => shallow(<Pagination {...props} />);

describe("Pagination component", () => {
  it("should render Pagination component", () => {
    const component = setUp({
      arrOfCurrButtons,
      currentPage,
      handlerBtnNext,
      handlerBtnPrev,
      handleClick,
    });
    expect(component).toMatchSnapshot();
  });
});
