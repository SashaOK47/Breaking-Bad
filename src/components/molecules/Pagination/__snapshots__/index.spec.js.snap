// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render Pagination component 1`] = `
<div
  className="pagination"
  data-test="pagination"
>
  <div
    className="paginationList"
  >
    <button
      className="paginationItem active"
      key="0"
      onClick={[MockFunction]}
    >
      1
    </button>
    <button
      className="paginationItem"
      key="1"
      onClick={[MockFunction]}
    >
      2
    </button>
    <button
      className="paginationItem"
      key="2"
      onClick={[MockFunction]}
    >
      3
    </button>
    <button
      className="paginationItem disabled"
      key="3"
      onClick={[MockFunction]}
    >
      ...
    </button>
  </div>
  <button
    className="btnPrev"
    onClick={[MockFunction]}
  >
    <img
      alt="Button Prev"
      src="test-file-stub"
    />
  </button>
  <button
    className="btnNext"
    onClick={[MockFunction]}
  >
    <img
      alt="Button Next"
      src="test-file-stub"
    />
  </button>
</div>
`;
